In [1]:
ls

00. _wd - working day.txt
01. KRXworkingDay_from1980.csv
CSV/
Currency/
Currency_describe_directory_list.ipynb
KOR/
KRX반도체_SK하이닉스_KOSPI200.xlsx
Korean_stock_describe.ipynb
README.md
Ridge_baseline.ipynb
Taiwan/
Taiwan_describe_directory_list.ipynb
US/
US_stock_describe_list_final.ipynb
_Rhistory
bloomberg_dataprocessing-Copy1.ipynb
bloomberg_dataprocessing.ipynb
bloomberg_smc_describe.xlsx
data_change_ver0.1.csv
data_change_ver0.2.csv
df_test.csv
df_test2.csv
economic_smc.csv
eda.ipynb
eda_from_change.ipynb
eda_var_name.csv
lasso_baseline.ipynb
list
list.csv
logistic_classifier_baseline.ipynb
new/
ols_baseline_regression.ipynb
preprocessing.ipynb
preprocessing.py
semiconductor_bloomberg_preprocessing.ipynb
semiconductor_bloomberg_preprocessing_selectedby_longdays.ipynb
semiconductor_feature_list_bloomberg.xlsx
total_data.csv
totaldata_preprocessing.ipynb
tree_based_regressor_baseline.ipynb
~$bloomberg_smc_describe.xlsx


In [2]:
pwd

'/Users/mokpolar/Documents/GitHub/KRX_semiconductor_index_prediction'

In [21]:
import pandas as pd
import re
%matplotlib inline

In [22]:
eco = pd.read_csv('/Users/mokpolar/Downloads/bloomberg_raw/economic.csv')

In [23]:
eco.set_index('col0', inplace = True)

In [24]:
eco.head()

col1  col2        col3  col4
col0                                        
GDP CQOQ Index     0     1  03/31/1981   8.1
GDP CQOQ Index     0     1  06/30/1981  -2.9
GDP CQOQ Index     0     1  09/30/1981   4.9
GDP CQOQ Index     0     1  12/31/1981  -4.3
GDP CQOQ Index     0     1  03/31/1982  -6.1

In [25]:
eco = eco.loc[['FDTR Index',
'USGG3M Index',
'USGG10YR Index',
'USTW$ Index',
'INJCJC Index',
'INJCSP Index',
'USEMNCHG Index',
'NFP PCH Index',
'NFP TCH Index',
'USMMMNCH Index',
'OUTFGAF Index',
'KOIMTOTY Index',
'KOFETOT% Index',
'KOEXTOTY Index',
'KOCPIMOM Index',
'SKCIMOM Index',
'USHETOT% Index',
'KOCPIYOY Index',
'SKCIYOY Index',
'USHEYOY Index',
'USURTOT Index',
'KOFETOT Index',
'SAARTOTL Index',
'NAPMPRIC Index',
'KOCPI Index',
'CONCCONF Index',
'NAPMNEWO Index',
'SKCITTL Index',
'CHPMINDX Index',
'USWHTOT Index',
'NAPMPMI Index',
'SKBSIC Index',
'CONSSENT Index',
'USERTOT Index',
'LEI WKIJ Index',
'KOEXTOT Index',
'KOIMTOT Index',
'NFP T Index',
'USEMTOT Index',
'KOIVCONY Index',
'KOEXPTIY Index',
'KOIMPTIY Index',
'KOIPMY Index',
'KOIPIY Index',
'KOPSIY Index',
'LEI YOY Index',
'KOPIIY Index',
'IP  YOY Index',
'KOIPOPSM Index',
'KOEXPTIM Index',
'KOIPMSM Index',
'KOIPIMOM Index',
'TMNOCHNG Index',
'KOIPMCY Index',
'KOIMPTIM Index',
'LEI IRTE Index',
'PITLYOY Index',
'CFNAI Index',
'PITLCHNG Index',
'KOPPIYOY Index',
'IP  CHNG Index',
'CFNAIMA3 Index',
'LEI CHNG Index',
'LEI ACE Index',
'SKLILY Index',
'KOPPIMOM Index',
'CPI YOY Index',
'CPI CHNG Index',
'PCE DEFY Index',
'CPUPXCHG Index',
'CPI XYOY Index',
'PCE CYOY Index',
'PIDSDPS Index',
'KOIPMS Index',
'KOIPISA Index',
'SKLILI Index',
'SKLILC Index',
'SKLICI Index',
'KOIPMC Index',
'KOIMPTI Index',
'LEI AVGW Index',
'KOPPI Index',
'IP Index',
'LEI TOTL Index',
'KOEXPTI Index',
'CPTICHNG Index',
'LEI STKP Index',
'LEI BP Index',
'PIDSPINX Index',
'LEI MNO Index',
'LEI NWCN Index',
'KOBPCB Index',
'KOBPFIN Index',
'KOBPTB Index',
'KODIBAL Index',
'FRNTTNET Index',
'FRNTTOTL Index',
'KOBPCA Index',
'KOMSMBY Index',
'CICRTOT Index',
'KOMSM1FY Index',
'KOMSM1Y Index',
'MTIBCHNG Index',
'MGT2TB Index',
'MGT2RE Index',
'OEUSKLAP Index',
'OEUSKLAR Index',
'OEKRN022 Index',
'GPDITOC% Index',
'KOECGCPY Index',
'KOECIMPY Index',
'KOECGCSY Index',
'KOECSIMQ Index',
'KOECPRCY Index',
'KOECSPRQ Index',
'KOECFCSY Index',
'KOECFCOY Index',
'KOECSEMQ Index',
'GDP CQOQ Index',
'GPGSTOC% Index',
'KOECEXPY Index',
'GDP CYOY Index',
'GDPCTOT% Index',
'GDP CURY Index',
'KOECSGVQ Index',
'KOECGVTY Index',
'GDP PIQQ Index',
'GDPCPCEC Index',
'GDP CUR$ Index',
'USCABAL Index',
'JNVNIYOY Index',
'COSTNFR% Index',
'KOGNICNY Index',
'KODFTOTY Index',
'KOGNICUY Index',
'EHCAUS Index',
'KOGCGDPY Index',
'PRODNFR% Index',
'KOGCSTOQ Index'], :]

In [33]:
eco.head()

col0  col1  col2        col3 col4
0  FDTR Index     0     1  01/01/1981   18
1  FDTR Index     0     1  01/02/1981   18
2  FDTR Index     0     1  01/05/1981   18
3  FDTR Index     0     1  01/06/1981   18
4  FDTR Index     0     1  01/07/1981   18

In [27]:
eco.reset_index(inplace = True)

In [30]:
len(eco)

93701

In [29]:
eco.to_csv('bloomberg_smc_longdays.csv')

In [34]:
eco.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 93701 entries, 0 to 93700
Data columns (total 5 columns):
col0    93701 non-null object
col1    93701 non-null int64
col2    93701 non-null int64
col3    93701 non-null object
col4    93701 non-null object
dtypes: int64(2), object(3)
memory usage: 3.6+ MB


### --- 여기까지는 Column 뽑아내기 ---

In [236]:
eco = pd.read_csv('bloomberg_smc_longdays.csv.csv')

In [35]:
len(eco)

93701

In [36]:
eco.head()

col0  col1  col2        col3 col4
0  FDTR Index     0     1  01/01/1981   18
1  FDTR Index     0     1  01/02/1981   18
2  FDTR Index     0     1  01/05/1981   18
3  FDTR Index     0     1  01/06/1981   18
4  FDTR Index     0     1  01/07/1981   18

In [37]:
eco['col4'] = pd.to_numeric(eco['col4'])

In [38]:
eco.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 93701 entries, 0 to 93700
Data columns (total 5 columns):
col0    93701 non-null object
col1    93701 non-null int64
col2    93701 non-null int64
col3    93701 non-null object
col4    93701 non-null float64
dtypes: float64(1), int64(2), object(2)
memory usage: 3.6+ MB


In [39]:
eco['col4'].sort_values()

88915     -9865.3
88956     -9411.8
88916     -5026.2
88957     -4051.9
88917     -3971.8
           ...   
57923    157288.0
57924    157878.0
81360    158103.0
57925    158269.0
57926    158510.0
Name: col4, Length: 93701, dtype: float64

In [40]:
# col3 type datetime으로 만들기 
eco['col3'] = pd.to_datetime(eco['col3'])

In [41]:
# 변수 종류는 289개
len(eco['col0'].unique())

139

In [42]:
# 변수별로 데이터 갯수 확인 필요 뭐야 이거 시박 다 다르네
for i, j in enumerate(eco['col0'].unique()):
    print(i, eco['col0'].unique()[i], len(eco[eco['col0'] == j]))    

0 FDTR Index 9747
1 USGG3M Index 9786
2 USGG10YR Index 9790
3 USTW$ Index 9591
4 INJCJC Index 1983
5 INJCSP Index 1982
6 USEMNCHG Index 456
7 NFP PCH Index 456
8 NFP TCH Index 456
9 USMMMNCH Index 456
10 OUTFGAF Index 456
11 KOIMTOTY Index 456
12 KOFETOT% Index 456
13 KOEXTOTY Index 456
14 KOCPIMOM Index 456
15 SKCIMOM Index 456
16 USHETOT% Index 456
17 KOCPIYOY Index 456
18 SKCIYOY Index 456
19 USHEYOY Index 456
20 USURTOT Index 456
21 KOFETOT Index 456
22 SAARTOTL Index 456
23 NAPMPRIC Index 456
24 KOCPI Index 456
25 CONCCONF Index 456
26 NAPMNEWO Index 456
27 SKCITTL Index 456
28 CHPMINDX Index 456
29 USWHTOT Index 456
30 NAPMPMI Index 456
31 SKBSIC Index 456
32 CONSSENT Index 456
33 USERTOT Index 456
34 LEI WKIJ Index 456
35 KOEXTOT Index 456
36 KOIMTOT Index 456
37 NFP T Index 456
38 USEMTOT Index 456
39 KOIVCONY Index 455
40 KOEXPTIY Index 455
41 KOIMPTIY Index 455
42 KOIPMY Index 455
43 KOIPIY Index 455
44 KOPSIY Index 455
45 LEI YOY Index 455
46 KOPIIY Index 455
47 IP  YOY Inde

In [46]:
k = pd.DataFrame()

In [47]:
k_1 = []

In [49]:
# 변수별로 데이터 갯수 확인해서 데이터 프레임으로 만들어 보자 
for i, j in enumerate(eco['col0'].unique()):
    k_1.append([i, eco['col0'].unique()[i], len(eco[eco['col0'] == j])])

In [53]:
import numpy as np

In [54]:
k_1.to_ndarray(column= ['index', 'ticker', 'len_data'])

AttributeError: 'list' object has no attribute 'to_ndarray'

In [56]:
npa = np.asarray(k_1)

In [59]:
k = pd.DataFrame(npa, columns = ['index', 'ticker', 'len_data'])

In [63]:
k.drop('index', axis = 1, inplace = True)

In [65]:
k.head()

ticker len_data
0      FDTR Index     9747
1    USGG3M Index     9786
2  USGG10YR Index     9790
3     USTW$ Index     9591
4    INJCJC Index     1983

In [66]:
k['len_data'].sort_values()

138     151
129     151
130     151
131     151
132     151
       ... 
30      456
3      9591
0      9747
1      9786
2      9790
Name: len_data, Length: 139, dtype: object

In [71]:
k['len_data'].unique()

array(['9747', '9786', '9790', '9591', '1983', '1982', '456', '455',
       '454', '452', '152', '151'], dtype=object)

데이터 갯수가 152, 151은 분기별 데이터

In [67]:
k.loc[138, :]

ticker      KOGCSTOQ Index
len_data               151
Name: 138, dtype: object

In [72]:
k.loc[30, :]

ticker      NAPMPMI Index
len_data              456
Name: 30, dtype: object

In [68]:
k.to_csv('bloom_smc_longdays_len_data.csv')

In [69]:
eco.head()

col0  col1  col2       col3  col4
0  FDTR Index     0     1 1981-01-01  18.0
1  FDTR Index     0     1 1981-01-02  18.0
2  FDTR Index     0     1 1981-01-05  18.0
3  FDTR Index     0     1 1981-01-06  18.0
4  FDTR Index     0     1 1981-01-07  18.0

In [70]:
eco[eco['col0'] == 'KOGCSTOQ Index']

col0  col1  col2       col3  col4
93550  KOGCSTOQ Index     0     1 1981-03-31   5.2
93551  KOGCSTOQ Index     0     1 1981-06-30   7.6
93552  KOGCSTOQ Index     0     1 1981-09-30   5.1
93553  KOGCSTOQ Index     0     1 1981-12-31   4.5
93554  KOGCSTOQ Index     0     1 1982-03-31   1.3
...               ...   ...   ...        ...   ...
93696  KOGCSTOQ Index     0     1 2018-06-30   1.1
93697  KOGCSTOQ Index     0     1 2018-09-30   1.1
93698  KOGCSTOQ Index     0     1 2018-12-31  -0.3
93699  KOGCSTOQ Index     0     1 2019-03-31  -0.8
93700  KOGCSTOQ Index     0     1 2019-06-30   1.5

[151 rows x 5 columns]

일단 기간을 1981뇬 3월 31일까지 있는거부터 2019년 6월 30일 까지 있는 티커들만 뽑았음  
근데 보니까 데이터 151개만 있는건 분기별 데이터네


In [73]:
eco[eco['col0'] == 'NAPMPMI Index']

col0  col1  col2       col3  col4
53823  NAPMPMI Index     0     1 1981-01-31  49.2
53824  NAPMPMI Index     0     1 1981-02-28  48.8
53825  NAPMPMI Index     0     1 1981-03-31  49.6
53826  NAPMPMI Index     0     1 1981-04-30  51.6
53827  NAPMPMI Index     0     1 1981-05-31  53.5
...              ...   ...   ...        ...   ...
54274  NAPMPMI Index     0     1 2019-06-30  51.7
54275  NAPMPMI Index     0     1 2019-07-31  51.2
54276  NAPMPMI Index     0     1 2019-08-31  49.1
54277  NAPMPMI Index     0     1 2019-09-30  47.8
54278  NAPMPMI Index     0     1 2019-10-31  48.3

[456 rows x 5 columns]

455, 456개 있는건 월별 데이터

In [74]:
k.loc[3, :]

ticker      USTW$ Index
len_data           9591
Name: 3, dtype: object

In [75]:
eco[eco['col0'] == 'USTW$ Index']

col0  col1  col2       col3     col4
29323  USTW$ Index     0     1 1981-01-02  95.6833
29324  USTW$ Index     0     1 1981-01-05  94.4815
29325  USTW$ Index     0     1 1981-01-06  94.3829
29326  USTW$ Index     0     1 1981-01-07  94.7921
29327  USTW$ Index     0     1 1981-01-08  95.1600
...            ...   ...   ...        ...      ...
38909  USTW$ Index     0     1 2019-10-25  92.0086
38910  USTW$ Index     0     1 2019-10-28  91.9877
38911  USTW$ Index     0     1 2019-10-29  91.8727
38912  USTW$ Index     0     1 2019-10-30  92.0533
38913  USTW$ Index     0     1 2019-10-31  91.7069

[9591 rows x 5 columns]

9591개 있는 건 데일리 데이터네 그럼 분류는 일, 월, 분기 세 가지 분류로 잡고   
월을 선으로 이어서 만들어 보자 

In [76]:
eco.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 93701 entries, 0 to 93700
Data columns (total 5 columns):
col0    93701 non-null object
col1    93701 non-null int64
col2    93701 non-null int64
col3    93701 non-null datetime64[ns]
col4    93701 non-null float64
dtypes: datetime64[ns](1), float64(1), int64(2), object(1)
memory usage: 3.6+ MB


인덱스 별로 데이터 프레임을 발라내자 !

In [78]:
list(eco['col0'].unique())

['FDTR Index',
 'USGG3M Index',
 'USGG10YR Index',
 'USTW$ Index',
 'INJCJC Index',
 'INJCSP Index',
 'USEMNCHG Index',
 'NFP PCH Index',
 'NFP TCH Index',
 'USMMMNCH Index',
 'OUTFGAF Index',
 'KOIMTOTY Index',
 'KOFETOT% Index',
 'KOEXTOTY Index',
 'KOCPIMOM Index',
 'SKCIMOM Index',
 'USHETOT% Index',
 'KOCPIYOY Index',
 'SKCIYOY Index',
 'USHEYOY Index',
 'USURTOT Index',
 'KOFETOT Index',
 'SAARTOTL Index',
 'NAPMPRIC Index',
 'KOCPI Index',
 'CONCCONF Index',
 'NAPMNEWO Index',
 'SKCITTL Index',
 'CHPMINDX Index',
 'USWHTOT Index',
 'NAPMPMI Index',
 'SKBSIC Index',
 'CONSSENT Index',
 'USERTOT Index',
 'LEI WKIJ Index',
 'KOEXTOT Index',
 'KOIMTOT Index',
 'NFP T Index',
 'USEMTOT Index',
 'KOIVCONY Index',
 'KOEXPTIY Index',
 'KOIMPTIY Index',
 'KOIPMY Index',
 'KOIPIY Index',
 'KOPSIY Index',
 'LEI YOY Index',
 'KOPIIY Index',
 'IP  YOY Index',
 'KOIPOPSM Index',
 'KOEXPTIM Index',
 'KOIPMSM Index',
 'KOIPIMOM Index',
 'TMNOCHNG Index',
 'KOIPMCY Index',
 'KOIMPTIM Index',
 'L

In [82]:
for i in list(eco['col0'].unique()):
    i = eco[eco['col0'] == i]

In [83]:
i

col0  col1  col2       col3  col4
93550  KOGCSTOQ Index     0     1 1981-03-31   5.2
93551  KOGCSTOQ Index     0     1 1981-06-30   7.6
93552  KOGCSTOQ Index     0     1 1981-09-30   5.1
93553  KOGCSTOQ Index     0     1 1981-12-31   4.5
93554  KOGCSTOQ Index     0     1 1982-03-31   1.3
...               ...   ...   ...        ...   ...
93696  KOGCSTOQ Index     0     1 2018-06-30   1.1
93697  KOGCSTOQ Index     0     1 2018-09-30   1.1
93698  KOGCSTOQ Index     0     1 2018-12-31  -0.3
93699  KOGCSTOQ Index     0     1 2019-03-31  -0.8
93700  KOGCSTOQ Index     0     1 2019-06-30   1.5

[151 rows x 5 columns]

In [ ]:
# input 은 dataframe

def index_slicer(dataframe, col_name):
    index_name = list(dataframe[col_name].unique())
    for i in index_name:
         = dataframe[dataframe[col_name] == i]
        
    return 생성된 수많은 데이터프레임      